addons/l10n_es_edi_tbai/models/account_move.py

```diff
@@ -0,0 +1,254 @@
+        "Valor Añadido. Artículo 80. Modificación de la base imponible.",
+        copy=False,
+    )
+
+    # -------------------------------------------------------------------------
+    # COMPUTE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('move_type', 'company_id')
+    def _compute_l10n_es_tbai_is_required(self):
+        for move in self:
+            move.l10n_es_tbai_is_required = move.is_sale_document() \
+                and move.country_code == 'ES' \
+                and move.company_id.l10n_es_tbai_tax_agency
+
+    @api.depends('company_id', 'edi_state', 'l10n_es_tbai_is_required', 'l10n_es_tbai_post_xml_id')
+    def _compute_l10n_es_tbai_id(self):
+        for record in self:
+            if record.l10n_es_tbai_is_required:
+                # TODO (POS): pre-sign document and store it as EDI document (before posting), only then get signature
+                # rationale: signature changes every time doc is signed, but needs to be consistent across XMLs / TBAI IDs
+                # cons: if pre-computed values need to be changed, signature will be wrong, QR won't work
+                if not record.l10n_es_tbai_signature:
+                    record.l10n_es_tbai_id = ''
+                else:
+                    company = record.company_id
+                    tbai_id_no_crc = '-'.join([
+                        'TBAI',
+                        str(company.vat[2:] if company.vat.startswith('ES') else company.vat),
+                        datetime.strftime(record.l10n_es_tbai_registration_date, '%d%m%y'),
+                        record.l10n_es_tbai_signature[:13],
+                        ''  # CRC
+                    ])
+                    record.l10n_es_tbai_id = tbai_id_no_crc + self._crc8(tbai_id_no_crc)
+            else:
+                record.l10n_es_tbai_id = ''  # record
+
+    @api.depends('l10n_es_tbai_id')
+    def _compute_l10n_es_tbai_qr(self):
+        for record in self:
+            if record.l10n_es_tbai_is_required and record.edi_state != 'to_send':
+                company = record.company_id
+                tbai_qr_no_crc = company.get_l10n_es_tbai_url_qr() + '?' + '&'.join([
+                    'id=' + record.l10n_es_tbai_id,
+                    's=' + record.l10n_es_tbai_sequence,
+                    'nf=' + record.l10n_es_tbai_number,
+                    'i=' + record._get_l10n_es_tbai_values_from_xml({'importe': r'.//ImporteTotalFactura'})['importe']
+                ])
+                record.l10n_es_tbai_qr = tbai_qr_no_crc + '&cr=' + self._crc8(tbai_qr_no_crc)
+            else:
+                record.l10n_es_tbai_qr = ''
+
+    @api.depends('l10n_es_tbai_qr')
+    def _compute_l10n_es_tbai_qr_escaped(self):
+        for record in self:
+            record.l10n_es_tbai_qr_escaped = url_quote(record.l10n_es_tbai_qr)
+
+    def _update_l10n_es_tbai_submitted_xml(self, xml_doc, cancel):
```

Comment:
No truth value is more appropriate or really makes sense to me as a default.

Reply:
I did use the default `cancel=False` in
- `TicketBaiWebServices` (multiple functions)
- `accound_edi_format._get_l10n_es_tbai_invoice_xml()`
- `account_move._get_l10n_es_tbai_submitted_xml()`

But I realize I only did that to save myself writing the argument in the code. Is that reason enough ?
If so I'll add it as a default there, otherwise I'll remove it as a default everywhere.